common/workunit/workunit.cpp

```diff
@@ -6342,6 +6347,50 @@ void CLocalWorkUnit::setSecurityToken(const char *value)
     p->setProp("@token", value);
 }
 
+bool CLocalWorkUnit::validateSecuritySignature() const
+{
+    CriticalBlock block(crit);
+    IDigitalSignatureManager * pDSM = createDigitalSignatureManagerInstanceFromEnv();
+    if (pDSM && pDSM->isDigiVerifierConfigured())
+    {
+        VStringBuffer wuSig("%s;%s", queryWuid(), queryUser());
+        VStringBuffer b64Signature("%s",p->queryProp("@signature"));
+        return  pDSM->digiVerify(wuSig, b64Signature);
+    }
+    WARNLOG("Unable to validate signature, please configure HPCCPublicKeyFile");
+    return false;
+}
+
+bool CLocalWorkUnit::setSecuritySignature()
+{
+    CriticalBlock block(crit);
+    IDigitalSignatureManager * pDSM = createDigitalSignatureManagerInstanceFromEnv();
+    if (pDSM && pDSM->isDigiSignerConfigured())
+    {
+        const char * wuid = queryWuid();
+        const char * user = queryUser();
+        if (isEmptyString(user))
+        {
+            ERRLOG("Cannot sign workunit, wuid not present");
+            return false;
+        }
+        if (isEmptyString(user))
+        {
+            ERRLOG("Cannot sign workunit, user ID not present");
+            return false;
+        }
+        VStringBuffer wuSig("%s;%s", wuid, user);
+        StringBuffer b64Signature;
+        if (pDSM->digiSign(wuSig, b64Signature))
```

Comment:
Is it best practice to include a random salt in the item being signed?  If so the digiVerify api doesn't really support that - or is that built into the signing/verify layer?